src/components/Card.js

```diff
@@ -1,31 +1,36 @@
+import { useState } from "react";
+
 import "./Card.css";
 
 function Card() {
-  const checkboxHandler = (event) => {
-    const card = document.querySelector('.card');
-    const caption = document.querySelector('.caption');
-    const text = document.querySelector('.text');
+  const [cardClass, setCardClass] = useState('card');
+  const [captionClass, setCaptionClass] = useState('caption');
+  const [textClass, setTextClass] = useState('text');
+  const [lineColor, setLineColor] = useState('#3f3f3f');
 
+  const checkboxHandler = (event) => {
     if (event.target.checked) {
-      card.classList.add('darkCard');
-      caption.classList.add('darkCaption');
-      text.classList.add('darkText');
+      setCardClass('card darkCard');
+      setCaptionClass('caption darkCaption');
+      setTextClass('text darkText');
+      setLineColor('#C0C0C0');
     } else {
-      card.classList.remove('darkCard');
-      caption.classList.remove('darkCaption');
-      text.classList.remove('darkText');
+      setCardClass('card');
+      setCaptionClass('caption');
+      setTextClass('text');
+      setLineColor('#3f3f3f');
     }
```

Comment:
Firstly, you could made a single state param for all of this. 'checked', for example, can control the classes on elements via ternary operator. (className={checked ? 'some classes' : 'some another classes'}). You can try to use 'classnames' library for a bit more comfort.
Secondly, it would be better to pass 'checked' from outside (for example in card object, that is in the list) instead of storing it in the Card's state.